main.cpp

```diff
@@ -18,13 +19,68 @@ using namespace clang::ast_matchers;
 using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
+private:
+    Rewriter* rewriter__=nullptr;
+
 public:
     CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
+        this->rewriter__ = &rewriter;
     };
 
-    void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
+    void run(const MatchFinder::MatchResult& Result) override {
+        const auto* C_STYLE =
+                Result.Nodes.getNodeAs<CStyleCastExpr>("cast");
+
+        if (!C_STYLE) {
+            return;
+        }
+
+        const int OFFSET_FOR_FROM = 1;
+        const int OFFSET_FOR_UP_TO = -1;
+        
+        SourceLocation range_from  =
+                C_STYLE->getLParenLoc().getLocWithOffset(OFFSET_FOR_FROM);
+
+        SourceLocation range_up_to =
+                C_STYLE->getRParenLoc().getLocWithOffset(OFFSET_FOR_UP_TO);
+
+        const auto TOKEN_RANGE =
+                CharSourceRange::getTokenRange(range_from, range_up_to);
+
+        const StringRef CAST_FROM = Lexer::getSourceText(TOKEN_RANGE,
+                                                        *Result.SourceManager,
+                                                        Result.Context->getLangOpts());
+
+       std::string CAST =
+                "static_cast<" + CAST_FROM.str() + ">";
+
+        const auto LOC =
+                C_STYLE->getSubExprAsWritten()->IgnoreImpCasts();
```

Comment:
What happens if the code is changed as follows:
```suggestion
                C_STYLE->getSubExpr()->IgnoreImpCasts();
```
What's the difference between the `getSubExprAsWritten()` and the `getSubExpr()` method?

Reply:
In my code I'm using it to be sure I'm working with raw expression case. So, the `getSubExprAsWritten` brings the exact row that we use in the test without any lexical optimizations.

Reply:
Could you provide an example of the difference between using these two methods?

Reply:
So, the same situation, don't have PC now to check, but in theory:
```c++
 ...
 int a = 5;
float b = 10.5;
a = b; 
// GetSubExpr() result is b = static_cast<float>(a). The class name of the node is ImpCastExpr, in this situation. 

ANOTHER (no imp casts)
int a = 5;
int b = a; 
// GetSubExpr() result will not change anything, thus no difference with GetSubExprAsWritten(). The claas name of the node is DecRefExpr
...
```

Reply:
Are you confident in the results of the examples you provided? Will the program change the code for such examples?